### ISSUE TITLE
emitCss broken in v2

### ISSUE THREAD

**Issue by Rich-Harris** [sveltejs/svelte-loader#56]
The `ast.hash` property was removed in v2, as we thought it wasn't being used anywhere (initially it was used for scoping styles, but then the scoping hash was generated from just the CSS rather than the entire component). 

Turns out it was being used in svelte-loader, to provide a filename for the temporary CSS file. So now we have a bundle of different components importing `svelte-undefined.css`, with predictable results — only one file matches that path.

A temporary workaround is to disable `emitCss`.

The easy fix would be to reinstate `ast.hash`, or (better) compute it here. But it would be nice to have a fix that also covered #45, which probably means writing the CSS files back into the source directory:

```
src
  .App.css
  .Thing.css
  App.html
  Thing.html
```

I don't know if there's a way to clean up those files after webpack is done with them (or create them in some 'virtual' form so they don't clutter the filesystem in the first place?

**Comment by mdebbar** [comment]
Emotion does something similar. It generates `<name>.emotion.css` files during the webpack build. My workaround was to delete the files in the build command:
```
package.json
{
  ...
  "scripts": {
    "build": "... && find . -name '*.emotion.css' | xargs -n 1 rm"
  }
}
```
but it would be nice to see this solved by webpack and loaders.

**Comment by kaisermann** [comment]
It seems that plugins can tap webpack's filesystem: [webpack-virtual-modules](https://github.com/sysgears/webpack-virtual-modules).

**Comment by kaisermann** [comment]
I've made a thing: https://github.com/kaisermann/svelte-loader/blob/fix-emitcss/index.js. It's working on the [example](https://github.com/kaisermann/template-webpack) I sent you over twitter.

The usage is something like: 

```js
// webpack.config.js
const svelteWebpack = require('svelte-loader')
module.exports = {
	...
	module: {
		rules: [
			{
				test: /\.html$/,
				exclude: /node_modules/,
				use: {
					loader: svelteWebpack.loader,
					options: {
						emitCss: true
					}
				}
			},
			{
				test: /\.css$/,
				use: ['css-loader']
			}
		]
	},
	plugins: [
		svelteWebpack.plugin,
	]
};
```

**Comment by cklmercer** [comment]
@kaisermann I tested your solution. Seems to work fairly well, however HMR requires a manual page refresh before styles will update.

**Comment by kaisermann** [comment]
I wasn't able to make it work with `watch` until [this commit on `webpack-virtual-modules`](https://github.com/sysgears/webpack-virtual-modules/commit/85208ef3c02e3ca419903484a93802c580d93e44) fixed the issue.

**Comment by kaisermann** [comment]
I wasn't able to make it work with `watch` until [this commit on `webpack-virtual-modules`](https://github.com/sysgears/webpack-virtual-modules/commit/85208ef3c02e3ca419903484a93802c580d93e44) fixed the issue.

By adding `hot: true` to the `svelte-webpack`'s `webpack.config.js` I was able to make it work with hot reloading as well.

**Comment by Rich-Harris** [comment]
Fixed in 2.9
